File: src/main/java/dev/cele/asa_sm/config/SpringApplicationContext.java
package dev.cele.asa_sm.config;

import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SpringApplicationContext implements ApplicationContextAware {
    private static ApplicationContext context;

    @Override
    public void setApplicationContext(ApplicationContext context) {
        SpringApplicationContext.context = context;
    }

    public static <T> T autoWire(Class<T> java) {
        return context.getBean(java);
    }

    @SuppressWarnings("unchecked")
    public static <T> T autoWire(String qualifier) {
        return (T) context.getBean(qualifier);
    }
}

File: src/main/java/dev/cele/asa_sm/config/SpringApplicationContext.kt
package dev.cele.asa_sm.config

import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.context.annotation.Configuration

@Configuration
internal open class SpringApplicationContext : ApplicationContextAware {

    override fun setApplicationContext(context: ApplicationContext) {
        Companion.context = context
    }

    companion object {
        @JvmStatic
        fun<T> autoWire(java: Class<T>): T {
            return context!!.getBean(java)
        }

        @JvmStatic
        fun<T> autoWire(qualifier: String): T {
            return context!!.getBean(qualifier) as T
        }

        @JvmStatic
        var context: ApplicationContext? = null
    }
}